test/suites/storage_driver_ceph.sh

```diff
@@ -129,7 +129,7 @@ test_storage_driver_ceph() {
     lxc profile device remove default root
     lxc storage delete "lxdtest-$(basename "${LXD_DIR}")-pool1"
     lxc storage delete "lxdtest-$(basename "${LXD_DIR}")-pool2"
-    ceph --cluster "${LXD_CEPH_CLUSTER}" osd pool rm "lxdtest-$(basename "${LXD_DIR}")-existing-osd-pool" "lxdtest-$(basename "${LXD_DIR}")-existing-osd-pool"  --yes-i-really-really-mean-it
+    ceph --cluster "${LXD_CEPH_CLUSTER}" osd pool rm "lxdtest-$(basename "${LXD_DIR}")-existing-osd-pool" "lxdtest-$(basename "${LXD_DIR}")-existing-osd-pool"  --yes-i-really-really-mean-it || true
```

Comment:
what do you mean by

"However, because it was the last line, the error was
not propagating."

If the pool is supposed to be removed by LXD, wouldn't it be better to change the check to confirm its really gone?

Reply:
I'm not that familiar with shell quirks, but for some reason when this line was the last line of the function,  its exit code being non-zero was not causing the function to exit with a non-zero exit code (despite `set -e`). 

However taking the main branch (so without the changes here) and adding any new execution following this one causes the function to exit appropriately at this line, and not reach the new line.

I'll update the line to instead check that the pool no longer exists.

Reply:
So... I tested this again, and despite the fact that I remember wasting a couple hours trying to figure out what was going on with this line, it seems the command no longer returns a non-zero exit code, even when it can't find an OSD pool to remove. 

Not sure if it got fixed with a ceph version update because it certainly was returning a non-zero exit code for a while...

Reply:
In any case, it's fixed now. I've replaced the removal attempt with a check that the osd pool was removed with the LXD storage pool.